### ISSUE TITLE
DeviceNDArray contiguity differs from Numpy array contiguity

### ISSUE THREAD

**Issue by gmarkall** [numba/numba#4943]
## Reporting a bug

- [X] I am using the latest released version of Numba (most recent is visible in
 the change log (https://github.com/numba/numba/blob/master/CHANGE_LOG).
- [x] I have included below a minimal working reproducer (if you are unsure how
 to write one see http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports).

Spotted whilst looking into #4628. Similar to #2766, but for `DeviceNDArrays`, the continuity disagrees with Numpy. I'm presently under the impression that this is a bug and not the result of some semantic difference when data is moved over to a CUDA device, so I'm looking into a fix to make them agree. The example code:

```python
from numba import cuda
import numpy as np

def test(shape, order):
    arr = np.ndarray(shape, order=order)
    d_arr = cuda.to_device(arr)
    print("Shape: %s, Order: %s" % (shape, order))
    print("Host:   C: %s, F: %s" % (arr.flags['C_CONTIGUOUS'], arr.flags['F_CONTIGUOUS']))
    print("Device: C: %s, F: %s" % (d_arr.flags['C_CONTIGUOUS'], d_arr.flags['F_CONTIGUOUS']))
    print()

test((1,4), 'C')
test((1,4), 'F')
test((4,1), 'C')
test((4,1), 'F')
```

produces:

```
Shape: (1, 4), Order: C
Host:   C: True, F: True
Device: C: True, F: False

Shape: (1, 4), Order: F
Host:   C: True, F: True
Device: C: False, F: True

Shape: (4, 1), Order: C
Host:   C: True, F: True
Device: C: True, F: False

Shape: (4, 1), Order: F
Host:   C: True, F: True
Device: C: False, F: True
```

**Comment by gmarkall** [comment]
... Actually this is the same as #2766, because the computation of flags is done by a `dummyarray.Array` backing the `DeviceNDArray`.

**Comment by gmarkall** [comment]
Originally I think the Numba and Numpy definitions of contiguity agreed. Then in Numpy 1.8, a relaxed definition of array contiguity was added: https://docs.scipy.org/doc/numpy/release.html#npy-relaxed-strides-checking

Since then, most people build with the relaxed definition enabled - see e.g. https://github.com/ContinuumIO/anaconda-issues/issues/117 (although that issue is still open, I'm using Numpy from Anaconda and it seems to have the relaxed definition - perhaps @asmeurer can confirm / close the Anaconda issue?)

I think the resolution here is to update Numba's contiguity checking to follow the relaxed definition. Numpy 1.7 will still use the old definition and probably disagree with Numba after that (though I haven't checked) but I don't consider it likely that making this change will cause an issue with 1.7, just as using the "strict" check in Numba with the relaxed check in Numpy doesn't seem to cause issues now.

Initially I've tried adjusting the logic in `numba.dummyarray.Array._compute_layout`  but I need to check that this doesn't cause knock-on issues, particularly in the CUDA target, e.g. with copying slices to/from device memory.
